File: core/src/galenscovell/oregontrail/graphics/CurrentLocationAnimation.java
package galenscovell.oregontrail.graphics;

import com.badlogic.gdx.graphics.g2d.*;
import galenscovell.oregontrail.util.*;

public class CurrentLocationAnimation {
    private TextureRegion sprite;
    private float frame, size;
    private int x, y;

    public CurrentLocationAnimation() {
        this.sprite = ResourceManager.uiAtlas.createSprite("current_marker");
        this.size = Constants.TILESIZE * 3;
        this.frame = 0.0f;
    }

    public void setTarget(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public void render(Batch batch) {
        batch.draw(sprite, x, y, size / 2, size / 2, size, size, 1, 1, -frame);
        if (frame < 360) {
            frame += 0.5f;
        } else {
            frame = 0;
        }
    }
}

File: core/src/galenscovell/hinterstar/graphics/CurrentLocationAnimation.scala
package galenscovell.hinterstar.graphics

import com.badlogic.gdx.graphics.g2d._
import galenscovell.hinterstar.util._


/**
  * CurrentLocationAnimation is an animation signifying the current Player location.
  */
class CurrentLocationAnimation {
  private final val sprite: TextureRegion = ResourceManager.uiAtlas.createSprite("current_marker")
  private final val size: Float = Constants.SECTORSIZE * 3
  private var frame: Float = 0.0f
  private var lx: Int = 0
  private var ly: Int = 0


  /**
    * Set the central location for this animation to revolve around.
    */
  def setTarget(x: Int, y: Int): Unit = {
    this.lx = x
    this.ly = y
  }

  /**
    * Render the animation revolving around the current location.
    */
  def render(batch: Batch): Unit = {
    batch.draw(sprite, lx, ly, size / 2, size / 2, size, size, 1, 1, -frame)
    if (frame < 360) {
      frame += 0.5f
    } else {
      frame = 0
    }
  }
}